src/components/interaction-designer/ResourceViewer.vue

```diff
@@ -23,10 +23,21 @@ export const DEBOUNCE_FLOW_PERSIST_MS = 1500
 export default {
   name: 'ResourceViewer',
   mixins: [Lang],
+  props: {
+    filteredBlocks: {
+      type: Array,
+      default: () => null,
+    },
+  },
   computed: {
-    ...mapGetters('resource-viewer', [
-      'filteredBlocks',
-    ]),
+    ...mapGetters({
+      activeFlow: 'flow/activeFlow',
+    }),
+    visibleBlocks() {
+      return Array.isArray(this.filteredBlocks)
+        ? this.filteredBlocks
+        : this.activeFlow?.blocks ?? []
```

Comment:
Resource Viewer will now display `filterdBlocks` from props if present, and all blocks from the `activeFlow` otherwise.

Reply:
`visibleBlocks` - good name 👍